### ISSUE TITLE
[NOTICE] The State of RHEL/CentOS 8.x and This Script

### ISSUE THREAD

**Issue by Zer0CoolX** [Zer0CoolX/guacamole-install-rhel#83]
There is too much info to pass along to those seeking a solution re: RHEL/CentOS 8.x, Apache Guacamole and this script to explain in the Projects | To-do List.

**TL;DR** - As of now, and until this issue is closed, the guac-install script is **NOT** compatible with RHEL/CentOS 8.x. Once it is feasible, I am going to revise the script which will most likely end up with its own repo (and will be linked to from this one). I am not neglecting updating the script for RHEL/CentOS 8.x, there are simply too many obstacles in the way right now that I have no control over.  For more details read on...

There is a web of conflicts that prevent progress on this matter for me:

1. RHEL/CentOS 8.x include `freerdp-2.x`
2. Guacamole 1.0.0 requires `freerdp 1.x`
3. Other packages required by Guacamole have been changed, renamed, replaced or even removed between RHEL/CentOS 7.x to 8.x. The most notable example of removal being `tomcat`. `libssh2-devel` was replaced by `libssh-devel`. Many other similar issues.

Apache Guacamole devs are working on updating Guacamole to work with `freerdp-2.x` which is expected to be working in Guacamole 1.1.0. However, as far as I am aware, there is no public ETA. It could be days, weeks, months or longer.

I could not find a safe, reliable way to install `freerdp-1.x` in RHEL/CentOS 8.x. so until Guacamole 1.1.0 is releases, issues 1 and 2 above essentially create a scenario I cannot resolve.

As for number 3 above, it will take time and effort to figure out alternative means to overcome issues caused by changes to other packages. Instead of fighting against them now, its smarter to wait until Guacamole 1.1.0 is released to see if it resolves any other issue (or creates new ones) in addition to the `freerdp` matter. I am a big fan of approaching things properly instead of trying to hack a solution together if it can be avoided.

So long story short, there are multiple reasons I have not made more swift progress on getting this script ready for RHEL/CentOS 8.x, trust me I wish it wasn't so. Essentially this leaves me waiting on Guacamole to release 1.1.0 before I am able to make further progress, test and release a working script.

Thank you.

PS - for reference, here are some resources on the matter:

- Dev talking about them working on fixing freerdp 2 and Guac compatibility. http://apache-guacamole-general-user-mailing-list.2363388.n4.nabble.com/Issue-with-Centos-7-7-and-FreeRDP-2-0-td6461.html
- The actual issue/report and progress for the freerdp matter. https://issues.apache.org/jira/browse/GUACAMOLE-249
- My very own [Issue 56](https://github.com/Zer0CoolX/guacamole-install-rhel/issues/56)

**Comment by volga629-1** [comment]
I can share SRPM latest release for fedora 29 server, that might will help you to easy the release for rhel family

**Comment by volga629-1** [comment]
http://ftpsrv01.networklab.ca/

**Comment by spyjdh** [comment]
Guacamole 1.1.0 has been released.
https://guacamole.apache.org/releases/1.1.0/

**Comment by Zer0CoolX** [comment]
Thanks for the heads up. I have been working on this and will continue to as time permits. I expect to have an updated version of the script for Guacamole 1.1.0 + RHEL/CentOS 7.x ready within the next few weeks. In the mean time hang tight.

Presuming Guacamole 1.1.0 and Freerdp 2.x play nice together, I will likely work on a new repo for Guacamole + RHEL/CentOS 8.x once I get the above ready. Stay tuned.

**Comment by Zer0CoolX** [comment]
This script is updated for Guac 1.1.0 and freerdp 2.x on RHEL/CentOS 7.7 and up. RHEL/CentOS 8.x will require re-writing a lot of code and working around new issues (like the repos no longer having tomcat built in).

When I am ready to support RHEL/CentOS 8.x I am likely to do so in a new repo and will link to it from this repo. I do not see this being ready soon, at least a few months.

**Comment by Makr91** [comment]
This may or may not help, but I have created a Packer  Project and Vagrant Cloud Image to go with a Vagrant  project using Ansible to setup Guacamole on CentOS 8.1. 

You can find the ansible files I used to provision the Packer image, Those were inspired by your script. Thank you!!

The Vagrant Project is more of a Quick Deployment tool, but it is the other half to the code to get a fully working instance.

I need to sanitize, and cleanup large portions of it but if you want to take a look it may help you get it working. 

So far I was able to get Telnet, RDP, SSH, Kubernetes and VNC working using FreeRDP2 on CentOS 8.1 using Tomcat 9 and Guacamole 1.1.0.

Packer Project:  https://github.com/Makr91/Packer-Guacamole-CentOS-8.1/tree/master

Specific Scripts that May help:  https://github.com/Makr91/Packer-Guacamole-CentOS-8.1/blob/master/ansible/main.yml

Vagrant Project: https://github.com/Makr91/Vagrant-Guacamole-CentOS-8.1

Specific Scripts that May help: https://github.com/Makr91/Vagrant-Guacamole-CentOS-8.1/blob/master/conf/ansible/Guac.yml

**Comment by Zer0CoolX** [comment]
Thank you for sharing, I will start looking at this and see what I can do to get this script ready for 8.x.

**Comment by lexcorp** [comment]
Does anyone have the libtelnet-devel library available for centos 8?

**Comment by adamboutcher** [comment]
Need to enable "CentOS Power Tools" for CentOS 8.
policycoreutils-python = policycoreutils-python-utils

Missing tomcat in CentOS8

**Comment by docop** [comment]
The script missing tomcat ? You do install manually or..

**Comment by adamboutcher** [comment]
> The script missing tomcat ? You do install manually or..

Sorry I wasnt clear enough, I was just writing down notes as I went along.
Apache tomcat as a package isnt available in CentOS 8, so the yum install line fails.
Same goes for libtelnet-devel.

**Comment by MarkProminic** [comment]
It's been several months since I've looked at my scripts/code and CentOS has updated to 8.2+.  I will try to take a look at this again this week and address Tomcat and Libtelnet-devel.

I am assuming that Adamboutcher and Docop are referring to my scripts?

**Comment by docop** [comment]
Yeah, and your are the best for sure ! hehe  I do get more free ressource to try out as well on my server. i will try the ssl install of your upcoming revision for sure.  Are you gonna do a last version for the centos7.x as well, before turning to the c8 ?

**Comment by adamboutcher** [comment]
> I am assuming that Adamboutcher and Docop are referring to my scripts?

No, I'm referring to Zer0CoolX's script. Mainly for others to see why it doesn't work in CentOS 8. 

I don't use ansible so unfortunately can't comment.
